extensions/amp-form/0.1/form-verifiers.js

```diff
@@ -0,0 +1,326 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dev} from '../../../src/log';
+import {
+  isJsonScriptTag,
+  scopedQuerySelector,
+  scopedQuerySelectorAll,
+} from '../../../src/dom';
+import {tryParseJson} from '../../../src/json';
+
+/** @visibleForTesting */
+export const CONFIG_KEY = 'verificationGroups';
+
+/**
+ * @typedef {{
+ *    name:string,
+ *    message:string
+ *  }}
+ */
+let VerificationErrorDef;
+
+/**
+ * @typedef {{
+ *    errors:!Array<!VerificationErrorDef>
+ *  }}
+ */
+let VerificationErrorResponseDef;
+
+/**
+ * Construct the correct form verifier based on whether
+ * a config block is present.
+ * @param {!HTMLFormElement} form
+ * @param {function():Promise<!../../../src/service/xhr-impl.FetchResponse>} xhr
+ */
+export function getFormVerifier(form, xhr) {
+  const configTag = getConfig_(form);
+  if (configTag) {
+    const config = dev().assert(parseConfig_(configTag));
+    return new AsyncVerifier(form, config, xhr);
+  } else {
+    return new DefaultVerifier(form, [], xhr);
+  }
+}
+
+/**
+ * @param {!HTMLFormElement} form
+ * @return {?Element}
+ */
+function getConfig_(form) {
+  return form.getElementsByTagName('script')[0];
+}
+
+/**
+ * @param {!Element} script
+ * @return {!Array<!VerificationGroup>}
+ */
+function parseConfig_(script) {
+  if (isJsonScriptTag(script)) {
+    const json = tryParseJson(script.textContent, () => {
+      throw new Error('Failed to parse amp-form config. Is it valid JSON?');
+    });
+    const config = json && json[CONFIG_KEY];
+    if (!config.length) {
```

Comment:
It's fine for this to be an error, but it _this_ line causes an error. As in, the test, not the error we throw on the line below.